openformats/formats/vtt.py

```diff
@@ -0,0 +1,177 @@
+                "don't follow '[start] --> [end] (position)' pattern"
+            )
+        try:
+            start = self._format_timing(start)
+        except ValueError:
+            raise ParseError(
+                f"Problem with start of timing at line {self.transcriber.line_number + 1}: '{start}'"
+            )
+        try:
+            end = self._format_timing(end)
+        except ValueError:
+            raise ParseError(
+                f"Problem with end of timing at line {self.transcriber.line_number + 1}: '{end}'"
+            )
+
+        # Content
+        string_to_translate = '\n'.join(src_strings[timings_index+1:])
+        if string_to_translate == "":
+            raise ParseError(f"Subtitle is empty on line {self.transcriber.line_number + 2}")
+
+        string = OpenString(timings, string_to_translate,
+                            occurrences=f"{start},{end}",
+                            order=next(self._order))
+        offset += len(identifier) + len(timings) + 1;
+        if len(identifier):
+            offset += 1
+        return offset, string
+
+    def _format_timing(self, timing):
+        try:
+            rest, milliseconds = timing.split('.')
+            milliseconds = f"{milliseconds:<03}"
+        except ValueError:
+            rest, milliseconds = timing, "000"
+        # timing may or may not contain hours part
+        if rest.count(':') == 1:
+            minutes, seconds = rest.split(':')
+            minutes, seconds, milliseconds = (int(minutes),
+                                              int(seconds),
+                                              int(milliseconds))
+            return f"{minutes:02}:{seconds:02}.{milliseconds:03}"
+        elif rest.count(':') == 2:
+            hours, minutes, seconds = rest.split(':')
+            hours, minutes, seconds, milliseconds = (int(hours),
+                                                    int(minutes),
+                                                    int(seconds),
+                                                    int(milliseconds))
+            return f"{hours:02}:{minutes:02}:{seconds:02}.{milliseconds:03}"
+        else:
+            raise ParseError(f"Unexpected timing format on line {self.transcriber.line_number + 2}")
+
+    def compile(self, template, stringset, **kwargs):
+        transcriber = Transcriber(template)
+        template = transcriber.source
+        stringset = iter(stringset)
+        string = next(stringset)
+
+        for start, subtitle_section in self._generate_split_subtitles(template):
+            transcriber.copy_until(start)
+            transcriber.mark_section_start()
+
+            # Find hash after timings
+            hash_position = -1
+            if subtitle_section.count('-->') > 0:
+                arrow_pos = subtitle_section.index('-->')
+                end_of_timings = subtitle_section.index('\n', arrow_pos + len('-->'))
+                hash_position = end_of_timings + 1
```

Comment:
How about:

```suggestion
            try:
                arrow_pos = subtitle_section.index('-->')
            except ValueError:
                hash_position = -1
            else:
                end_of_timings = subtitle_section.index('\n', arrow_pos + len('-->'))
                hash_position = end_of_timings + 1
```

I know mine is longer and this is a styling preference so feel free to ignore but I feel it is more _pythonic_.

Reply:
In current PR code, `.index('-->')` cannot produce an exception because it is under condition `(.count('-->') > 0)` in the previous line. 
But `.index('\n', arrow_pos + len('-->'))`  (in both current code and in your suggestion) can give an exception.
So in the new commit there is a rework.